src/targets/gpu/device/gelu.cpp

```diff
@@ -0,0 +1,60 @@
+#include <migraphx/gpu/device/gelu.hpp>
+#include <migraphx/gpu/device/nary.hpp>
+#include <migraphx/gpu/device/types.hpp>
+#include <cmath>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+// x * 0.5 * (1.0 + erf(x / sqrt(2.0)))
+template <class T>
+auto gelu_fn(T x) __device__
+{
+    return x * 0.5 * (1 + ::erf(x * M_SQRT1_2));
+}
+
+// 0.5 * x * (1 + tanh(sqrt(2 / pi) * (x + 0.044715 * pow(x, 3))))
+template <class T>
+auto gelu_fn_new(T x) __device__
+{
+    return 0.5 * x * (1 + tanh(sqrt(M_2_PI) * (x + 0.044715 * pow(x, 3))));
```

Comment:
In this case, I don't think it matters since gelu_fn() will be called instead by default. There is a slight accuracy penalty, but the performance is much better.

Reply:
I will change it anyways since the pow() function is much slower regardless.